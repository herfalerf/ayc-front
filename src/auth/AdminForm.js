import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../common/Alert";

// Admin Login form.
//
// Shows form and manages update to state on changes.
// On submission:
// - calls login function prop
// - redirects to /admin route
//
// Routes -> AdminForm -> Alert
// Routed as /login
//

function AdminForm({ login }) {
  const history = useHistory();
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    "AdminForm",
    "login=",
    typeof login,
    "formData=",
    formData,
    "formErrors",
    formErrors
  );

  // Handle form submit
  //
  // Calls login func prop and, if successful, redirect to /admin.
  //

  async function handleSubmit(evt) {
    evt.preventDefault();
    let result = await login(formData);
    if (result.success) {
      history.push("/admin/home");
    } else {
      setFormErrors(result.errors);
    }
  }

  // Update form data field
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((l) => ({ ...l, [name]: value }));
  }

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label>Username</label>
        <input
          name="username"
          className="form-control"
          value={formData.username}
          onChange={handleChange}
          autoComplete="username"
          required
        />
      </div>
      <div>
        <label>Password</label>
        <input
          type="password"
          name="password"
          className="form-control"
          value={formData.password}
          onChange={handleChange}
          autoComplete="current-password"
          required
        />
      </div>

      {formErrors.length ? <Alert type="danger" messages={formErrors} /> : null}

      <button onSubmit={handleSubmit}>Submit</button>
    </form>
  );
}

export default AdminForm;
