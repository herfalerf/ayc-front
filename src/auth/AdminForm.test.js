import React from "react";
import { render } from "@testing-library/react";
import AdminForm from "./AdminForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <AdminForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
