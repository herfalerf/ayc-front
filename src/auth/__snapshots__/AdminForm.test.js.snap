// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label>
        Username
      </label>
      <input
        autocomplete="username"
        class="form-control"
        name="username"
        required=""
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        autocomplete="current-password"
        class="form-control"
        name="password"
        required=""
        type="password"
        value=""
      />
    </div>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
